src/Store.cpp

```diff
@@ -9,97 +10,169 @@
-		cout << " 0) Exit" << endl << endl;
-		cout << " ---------------------------------- " << endl << endl;
-		cout << " What do you want to buy today?" << endl << endl << " ";
-		choice = input();
+		// Display the store's main menu
+		DisplayStoreMainMenu(_Player);
+
+		// Input choice
+		unsigned short choice = input();
 		cout << endl;
+
+		isValidChoice = true;
 		switch (choice)
 		{
-		case ITEMTYPE::ARROWS:
-			if (coins < arrowsCost)
-			{
-				cout << lowCoinsMessage << endl;
-			}
-			else
-			{
-				cout << " 5 Arrows purchased ! " << endl;
-				_Player->AddStoreItemToInventory(ITEMTYPE::ARROWS);
-				_Player->LoseCoins(arrowsCost);
-			}
-			Sleep(SLEEP_MS);
+		case 1: // Buy
+			BuyFront(_Player);
 			break;
-		case ITEMTYPE::BOMB:
-			if (coins < bombCost)
-			{
-				cout << lowCoinsMessage << endl;
-			}
-			else
-			{
-				cout << " Bomb purchased !" << endl;
-				_Player->AddStoreItemToInventory(ITEMTYPE::BOMB);
-				_Player->LoseCoins(bombCost);
-			}
-			Sleep(SLEEP_MS);
-			break;
-		case ITEMTYPE::POTION:
-			if (coins < potionCost)
-			{
-				cout << lowCoinsMessage << endl;
-			}
-			else
-			{
-				cout << " Potion purchased !" << endl;
-				_Player->AddStoreItemToInventory(ITEMTYPE::POTION);
-				_Player->LoseCoins(potionCost);
-			}
-			Sleep(SLEEP_MS);
+		case 2: // Sell
+			SellFront(_Player);
 			break;
-		case ITEMTYPE::WHETSTONE:
-			if (coins < whetstoneCost)
-			{
-				cout << lowCoinsMessage << endl;
-			}
-			else
-			{
-				cout << " Whetstone purchased !" << endl;
-				_Player->AddStoreItemToInventory(ITEMTYPE::WHETSTONE);
-				_Player->LoseCoins(whetstoneCost);
-			}
-			Sleep(SLEEP_MS);
-			break;
-		case 0://Exit Store
+		case 0: // Exit
 			break;
 		default:
 			cout << invalidChoice << endl;
+			isValidChoice = false;
 			Sleep(SLEEP_MS);
+			break;
+		}
+	} while (!isValidChoice);
+}
+
+void Store::BuyFront(Player* _Player) {
+	// Displays the buy front
+
+	const static catalogue_t stock_items = {
+		{ITEMTYPE::WHETSTONE, {"Whetstone", 1, 20}},
+		{ITEMTYPE::POTION, {"Potion", 1, 150}},
+		{ITEMTYPE::BOMB, {"Bomb", 1, 100}},
+		{ITEMTYPE::ARROWS, {"Arrows", 5, 40}}
+	};
+
+	while (true) {
+		ClearScreen();
+		cout << "\n\n";
+		_Player->DisplayInventory();
+		DisplayStock(stock_items);
+		cout << " What do you want to buy today?\n\n ";
+
+		int choice = input();
+		if (choice == 0 /* exit */) {
+			break;
+		}
+
+		const auto& item = stock_items.find(static_cast<ITEMTYPE>(choice));
+		if (item != std::end(stock_items)) {
+			BuyItem(_Player, static_cast<ITEMTYPE>(choice), item->second);
+		} 
+		else {
+			cout << invalidChoice << '\n';
+		}
+		Sleep(SLEEP_MS);
+	}
+}
+
+void Store::SellFront(Player* _Player) {
+	// Displays the sell front
+
+	const static catalogue_t stock_items = {
+		{ITEMTYPE::WHETSTONE, {"Whetstone", 1, 20}},
+		{ITEMTYPE::POTION, {"Potion", 1, 150}},
+		{ITEMTYPE::BOMB, {"Bomb", 1, 100}},
+		{ITEMTYPE::ARROWS, {"Arrows", 5, 40}}
+	};
```

Comment:
Could you make this a private member variable, also _not_ const? Reason is, as the game progresses new items may be added to the store.

Reply:
Do you want this private variable to be static as well across all instances of `Store`?

Reply:
Not static. Not making it static makes it possible for different stores to have different items, if we expand on the `Store` in the future.

Reply:
I'll merge this first, and if you would like to, you're certainly welcome to make a second PR for interacting with the stock. :)